File: src/org/usfirst/frc/team1678/robot/RobotParts.java
package org.usfirst.frc.team1678.robot;

import edu.wpi.first.wpilibj.AnalogInput;
import edu.wpi.first.wpilibj.DigitalInput;
import edu.wpi.first.wpilibj.Encoder;
import edu.wpi.first.wpilibj.VictorSP;

public class RobotParts {
	// All sensors are temporary values
	// TODO Change these when electrical wires things in
	public static Encoder yAxisEnc = new Encoder(0, 1);
	public static Encoder xAxisEnc = new Encoder(4, 5);
	
	public static DigitalInput xHallEffect = new DigitalInput(8);
	public static DigitalInput yHallEffect = new DigitalInput(9);
	public static DigitalInput extendHallEffect = new DigitalInput(11);
	
	public static AnalogInput grabberProxSensor = new AnalogInput(12);
	
	// Acutators
	public static VictorSP yAxisMotor1 = new VictorSP(3), yAxisMotor2;
	public static VictorSP xAxisMotor = new VictorSP(5);
	public static VictorSP extendMotor = new VictorSP(6);
	public static VictorSP grabberMotor1 = new VictorSP(0), grabberMotor2, grabberMotor3;
}

File: frc1678/RobotParts.cpp
#include "RobotParts.h"
#include "WPILib.h"

Encoder *RobotParts::YAxisEnc_;
Encoder *RobotParts::XAxisEnc_;

DigitalInput *RobotParts::XHallEffect_;
DigitalInput *RobotParts::YHallEffect_;
DigitalInput *RobotParts::ExtendHallEffect_;

AnalogInput *RobotParts::GrabberProxSensor_;

VictorSP *RobotParts::YAxisMotor1_;
VictorSP *RobotParts::YAxisMotor2_;
VictorSP *RobotParts::XAxisMotor_;
VictorSP *RobotParts::ExtendMotor_;
VictorSP *RobotParts::GrabberMotor1_;
VictorSP *RobotParts::GrabberMotor2_;
VictorSP *RobotParts::GrabberMotor3_;

void RobotParts::Initialize() {
  RobotParts::YAxisEnc_ = new Encoder(1, 2);
  RobotParts::XAxisEnc_ = new Encoder(8, 9);

  RobotParts::XHallEffect_ = new DigitalInput(6);
  RobotParts::YHallEffect_ = new DigitalInput(7);
  RobotParts::ExtendHallEffect_ = nullptr;

  RobotParts::GrabberProxSensor_ = new AnalogInput(0);

  RobotParts::YAxisMotor1_ = new VictorSP(6);
  RobotParts::YAxisMotor2_ = new VictorSP(7);
  RobotParts::XAxisMotor_ = new VictorSP(1);
  RobotParts::ExtendMotor_ = new VictorSP(0);
  RobotParts::GrabberMotor1_ = new VictorSP(3);
  RobotParts::GrabberMotor2_ = new VictorSP(5);
  RobotParts::GrabberMotor3_ = new VictorSP(8);
}
